clinica/iotools/bids_utils.py

```diff
@@ -49,6 +52,240 @@ class StudyName(str, Enum):
 }
 
 
+class BIDSSubjectID(ABC, UserString):
+    """This is the interface that BIDS subject IDs have to implement."""
+
+    def __init__(self, value: str):
+        instance = super().__init__(self.validate(value))
+        return instance
+
+    @abstractmethod
+    def validate(self, value: str) -> str:
+        raise NotImplementedError
+
+    @classmethod
+    @abstractmethod
+    def from_original_study_id(cls, study_id: str) -> str:
+        raise NotImplementedError
+
+    @abstractmethod
+    def to_original_study_id(self) -> str:
+        """Not sure we need that..."""
+        # todo : verify needed
+        raise NotImplementedError
+
+
+def _id_factory(study: StudyName) -> Type[BIDSSubjectID]:
```

Comment:
Looks like you are using this function outside the `bids_utils` module, so probably means it should be public. And maybe `bids_id_factory` ?

```suggestion
def bids_id_factory(study: StudyName) -> Type[BIDSSubjectID]:
```